generate_report_ui.gs

```diff
@@ -0,0 +1,263 @@
+    
+  if(input.timeFilter == "LAST_HOUR") { 
+    title = "Last Hour";
+    millisInBetween = 60*60*1000;
+  } else if(input.timeFilter == "LAST_6_HOUR") {
+    title = "Last 6 Hours";
+    millisInBetween = 6*60*60*1000;
+  } else if(input.timeFilter == "LAST_24_HOUR") {
+    title = "Last 24 Hours";
+    millisInBetween = 24*60*60*1000;
+  } else if(input.timeFilter == "LAST_7_DAYS") {
+    title = "Last 7 Days";
+    millisInBetween = 7*24*60*60*1000;
+  } else {
+    title = "Last 30 Days";
+    millisInBetween = 30*24*60*60*1000;
+  }
+  
+  if(projectFilter == "SPECIFIC_PROJECT") {
+    projTitle = "Specific Project";
+  } else if(projectFilter == "CUSTOM_PROJECT") {
+    projTitle = "Custom Projects";
+  } else if(projectFilter == "SYSTEM_PROJECT") {
+    projTitle = "System Projects";
+  } else {
+    projTitle = "All Projects";
+  }
+  
+  // variables to get the start time and end time
+  var fromTime = new Date(toTime.getTime() - millisInBetween);
+  var toTime = Utilities.formatDate(toTime, 'Etc/GMT', 'yyyy-MM-dd\'T\'HH:mm:ss.SSS\'Z\'');
+  var fromTime = Utilities.formatDate(fromTime, 'Etc/GMT', 'yyyy-MM-dd\'T\'HH:mm:ss.SSS\'Z\'');
+  
+  // if the selected option is specific project
+  if(projectFilter == "SPECIFIC_PROJECT") {
+    if(reportType == "mostExecutedScript") {
+      var processes = getNumberOfExecutionOfScript(input.projectId, fromTime);
+      var mostExecutedScript = convertObjectToSortedArrayForSpecificProject(processes, input.projectId);
+      var reportUrl = generateReportForMostExecutedScript(mostExecutedScript, "Report for Most Executed Script of "+ projTitle + " for " + title, limitResponse);
+    }
+    else if(reportType == "mostActiveUsers") {
+      var users = getUsersWithProcessId(input.projectId, fromTime);
+      var mostActiveUsers = convertObjectToSortedArrayForSPECIFIC_PROJECT(users, input.projectId);
+      var reportUrl = generateReportForMostActiveUsers(mostActiveUsers, "Report for Most Active Users of "+ projTitle + " for " + title, limitResponse);
+    }
+    else {
+      var emailOfOwnerOfScript = getUpdatedNameAndOwnerOfScript(input.projectId);
+      var emailOfOwnerOfScripts=[emailOfOwnerOfScript];
+      var reportUrl = generateReportForOwnerOfScripts(emailOfOwnerOfScripts, "Report for Owners of Apps Script of "+ projTitle);
+    } 
+  } else {
+    if(reportType == "mostExecutedScript") {
+      var mostExecutedScript = getMostExecutedScriptFromAllCloudProjects(fromTime , projectFilter);
+      var reportUrl = generateReportForMostExecutedScript(mostExecutedScript, "Report for Most Executed Script of "+ projTitle + " for " + title, limitResponse);
+    }
+    else if(reportType == "mostActiveUsers") {
+      var mostActiveUser = getMostActiveUser(fromTime , projectFilter);
+      var reportUrl = generateReportForMostActiveUsers(mostActiveUser, "Report for Most Active Users of "+ projTitle + " for " + title, limitResponse);
+    }
+    else
+    {
```

Comment:
{ braces has to be in same line as else or function

Reply:
Updated